test/built-ins/Array/prototype/fill/return-abrupt-from-this-length.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 22.1.3.6
+description: >
+  Return abrupt from ToLength(Get(O, "length")).
+info: >
+  22.1.3.6 Array.prototype.fill (value [ , start [ , end ] ] )
+
+  1. Let O be ToObject(this value).
+  2. ReturnIfAbrupt(O).
+  3. Let len be ToLength(Get(O, "length")).
+  4. ReturnIfAbrupt(len).
+---*/
+
+var o = {};
+
+Object.defineProperty(o, 'length', {
+  get: function() {
+    throw new Test262Error();
+  },
+  configurable: true
+});
+assert.throws(Test262Error, function() {
+  [].fill.call(o, 1);
+});
+
+delete o.length;
```

Comment:
You can simplify this test by creating a new object to use as the "this" value for the second invocation of `fill`